### PR TITLE
EAMxx: enable kokkos bounds checks in full_debug standalone build

### LINKED ISSUE #2828: are we not checking OOB?
This bug was causing an elusive (to Naser) segfault. The test was passing on pm-gpu just fine, but failed on other machines.

Are we not checking OOB as rigorously as we could?

              ```suggestion
        const int icol   = idx / nlevs;
```

ht @bartgol

_Originally posted by @mahf708 in https://github.com/E3SM-Project/scream/pull/2822#discussion_r1603644915_

### PR DESCRIPTION
**Pull request by bartgol**
Fixes #2828

### CODE REVIEW

File: components/eamxx/scripts/test_factory.py
Comment (jgfouca) [inline components/eamxx/scripts/test_factory.py:84]: Several of the other TestProperty subclasses use the DBG.CMAKE_ARGS list.

SP: `DBG.CMAKE_ARGS + [("SCREAM_DOUBLE_PRECISION", "False")],`
FPE: `DBG.CMAKE_ARGS + [("SCREAM_PACK_SIZE", "1"), ("SCREAM_FPE","True")]`

So this turns on bounds checking for DBG, SP, and FPE. Is that the intent?

Comment (bartgol) [reply components/eamxx/scripts/test_factory.py:84]: Ah, no I didn't want to do that. Although, should we? I def don't want it on in the valg build, so I need to edit that. I would argue that bounds can be checked in just one build, since access patterns do not depend on precision or FPE handling... I will make sure other builds don't inherit that.

Comment (jgfouca) [reply components/eamxx/scripts/test_factory.py:84]: Probably not. You can just explicitly disable it in the SP and FPE tests.

Comment (jgfouca) [reply components/eamxx/scripts/test_factory.py:84]: VALG does not use DBG.CMAKE_ARGS, so we are fine there.

**Comment by bartgol** [comment]
This PR brought up an issue in EKAT. I submitted a PR there, but we then need to wait for an upstream merge, since ekat will have kokkos 4.2 in it now. So this PR is on hold until E3SM-Project/E3SM#6421 is merged, we do an upstream merge, E3SM-Project/EKAT#339 is merged, and we update EKAT in the scream repo.

**Review by jgfouca — APPROVED** [review]

**Pull request merged**
